loader/loader.c

```diff
@@ -2850,6 +2888,43 @@ static VkResult loaderReadLayerJson(const struct loader_instance *inst, struct l
         goto out;
     }
 
+    props->num_blacklist_layers = 0;
+    props->blacklist_layer_names = NULL;
+    blacklisted_layers = cJSON_GetObjectItem(layer_node, "blacklisted_layers");
+    if (blacklisted_layers != NULL && strcmp(name, VK_OVERRIDE_LAYER_NAME)) {
```

Comment:
Wouldn't this (and the corresponding else if) be cleaner if you created a top level if and a lower if/else?  Otherwise, you're potentially running the NULL compare twice.

```cpp
if (blacklisted_layers != NULL) {
   if (strcmp(name, VK_OVERRIDE_LAYER_NAME)) {
   } else {
   }
}
```